### PR TITLE
Jacoco reports

### PR DESCRIPTION
**Pull request by karimodm**

### CODE REVIEW

File: iota-buildenv-ci-build-docker/pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+steps:
```
Comment (karimodm) [inline iota-buildenv-ci-build-docker/pipeline.yml]: Does this file even belong in the PR? What is its purpose in the context of Jacoco reports?

File: iri-regression-tests/fetch_info.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+from yaml import load, Loader
```
Comment (karimodm) [inline iri-regression-tests/fetch_info.py]: What's the use for this file? I don't manage to find references to it.

File: iri-regression-tests/pipeline.yml [hunk 3]
```diff
@@ -95,7 +105,7 @@ steps:
         ERROR=0
         for machine_dir in tests/features/machine?; do
           mkdir -p /cache/\$machine_dir/
-          python /cache/tiab/create_cluster.py -i iotacafe/iri-dev:$IRI_BUILD_NUMBER -t $BUILDKITE_BUILD_ID -c \$machine_dir/config.yml -o /cache/\$machine_dir/output.yml -k /conf/kube/kube.config -n buildkite -x /cache/iri/python-regression -e "apt update && apt install unzip && wget http://search.maven.org/remotecontent\?filepath\=org/jacoco/jacoco/0.8.4/jacoco-0.8.4.zip -O jacoco.zip && mkdir /opt/jacoco && unzip jacoco.zip -d /opt/jacoco" -d
```
Comment (karimodm) [inline iri-regression-tests/pipeline.yml]: If the change you are applying is a breaking change on depends on other functionality (like the `-e` flag here), don't push directly to `master`. It is good practice to make every significant change in a self-contained PR that can be submitted for review and treated, possibly, as an atomic change.

File: iri-regression-tests/pipeline.yml [hunk 4]
```diff
@@ -95,7 +105,7 @@ steps:
...
+          python /cache/tiab/create_cluster.py -i iotacafe/iri-dev:$IRI_BUILD_NUMBER -t $BUILDKITE_BUILD_ID -c \$machine_dir/config.yml -o /cache/\$machine_dir/output.yml -k /conf/kube/kube.config -n buildkite -x /cache/iri/python-regression -d
```
Comment (karimodm) [inline iri-regression-tests/pipeline.yml]: The `-e` parameter should be re-introduced to enable Jacoco functionality. So for the time being I consider this PR not to be complete.

File: iota-buildenv-ci-build-docker/pipeline.yml (hunk 1 above)
Comment (sadjy) [reply iota-buildenv-ci-build-docker/pipeline.yml]: No it did not belong there. I messed up. I added back to the master already.

File: iri-regression-tests/fetch_info.py (hunk 2 above)
Comment (sadjy) [reply iri-regression-tests/fetch_info.py]: As I said, this file is now useless as I did it on a different train of thoughts prior using kubectl to fetch podnames.

File: iri-regression-tests/pipeline.yml (hunk 3 above)
Comment (sadjy) [reply iri-regression-tests/pipeline.yml]: Makes sense. Noted.

File: iri-regression-tests/pipeline.yml (hunk 4 above)
Comment (sadjy) [reply iri-regression-tests/pipeline.yml]: Yep.

File: iri-push-docker-trusted/pipeline.yml [hunk 5]
```diff
@@ -36,3 +36,4 @@ steps:
         IRI_BUILD_NUMBER: $IRI_BUILD_NUMBER
         IRI_GIT_COMMIT: $IRI_GIT_COMMIT
         IRI_BUILDKITE_PULL_REQUEST_REPO: $IRI_BUILDKITE_PULL_REQUEST_REPO
+        ARTIFACT_BUILDKITE_BUILD_ID: $ARTIFACT_BUILDKITE_BUILD_ID
```
Comment (sadjy) [inline iri-push-docker-trusted/pipeline.yml]: The `jacoco report` command which generates the actual report from the merged `jacoco.exec` file requires `.java` classfiles in order to work. 
The plan is to upload those files as artifacts from the `iri build jar` pipeline. I added this line so we can download the classfiles as artifacts in the `iri-regression-tests` pipeline.
The necessary classfiles are yet to be identified.

File: iota-buildenv-ci-build-docker/pipeline.yml (hunk 1 above)
Comment (karimodm) [reply iota-buildenv-ci-build-docker/pipeline.yml]: You marked the conversation as resolved but the file is still in this PR. Should it be removed or should it stay? I don't understand what's the status of this file now.

File: iri-regression-tests/pipeline.yml [hunk 6]
```diff
@@ -128,8 +140,43 @@ steps:
...
+            if [ \$(kubectl --kubeconfig=/conf/kube/kube.config --namespace=buildkite get pod \$pod -o jsonpath='{.metadata.labels.tag}') = \$BUILDKITE_BUILD_ID ]
+            then
+              /cache/kubectl --kubeconfig=/conf/kube/kube.config --namespace=buildkite exec \$pod -- /bin/bash -c "pkill java"
+              /cache/kubectl --kubeconfig=/conf/kube/kube.config cp buildkite/\$pod:/iri/jacoco.exec /cache/jacoco/jacoco_\$pod.exec
+            fi
+          done
+        done
+      - apt update && apt install unzip && wget http://search.maven.org/remotecontent\?filepath\=org/jacoco/jacoco/0.8.4/jacoco-0.8.4.zip -O jacoco.zip && unzip jacoco.zip -d /cache/jacoco && rm jacoco.zip
+      - java -jar /cache/jacoco/lib/jacococli.jar merge /cache/*.exec --destfile /cache/jacoco.exec
+      - buildkite-agent artifact download target/iri-oracle8-${IRI_BUILD_NUMBER}.jar /cache/jacoco/ --build $ARTIFACT_BUILDKITE_BUILD_ID
```
Comment (karimodm) [inline iri-regression-tests/pipeline.yml]: Where is this artifact uploaded from? This will be a bit tricky to verify without running it.

Comment (karimodm) [reply iri-regression-tests/pipeline.yml]: Oh... I see
```
     artifact_paths:
       - "target/iri-oracle8-*"
```
in `iri-build-jar` and `iri-build-jar-prs`.

**Pull request merged**